controller/src/test/java/io/pravega/controller/server/ZKControllerServiceMainTest.java

```diff
@@ -91,105 +91,120 @@ protected void shutDown() throws Exception {
         }
     }
 
-    @Test(timeout = 100000)
+    /**
+     * This test verifies that the Controller mechanism for handling a ZK session expiration works correctly. Moreover,
+     * this verification is done both assuming that Zookeeper client connection parameters do not change and simulating
+     * a change on them.
+     */
+    @Test(timeout = 120000)
     public void testZKSessionExpiry() throws Exception {
```

Comment:
The changes to this test case are making it super long. We don't have to do it in this PR, as it would be good to unblock these changes, but I'd recommend creating an issue to improve it in the future and separate the test cases.

Reply:
Here is the issue: https://github.com/pravega/pravega/issues/3449